packages/elements/list/src/getListDeleteForward.ts

```diff
@@ -22,15 +30,52 @@ export const getListDeleteForward = (editor: SPEditor) => {
   const res = getListItemEntry(editor, {});
 
   let moved: boolean | undefined = false;
-  if (!isSelectionAtBlockEnd(editor) || !res) {
+  if (!isSelectionAtBlockEnd(editor)) {
     return moved;
   }
 
+  if (!res) {
+    const afterPoint = Editor.after(editor, editor.selection!.focus.path);
+
+    if (afterPoint) {
+      // there is a block after it
+      const nextSiblingListRes = getListItemEntry(editor, {
+        at: afterPoint.path,
+      });
+
+      if (nextSiblingListRes) {
+        // the next block is a list
+        const { listItem } = nextSiblingListRes;
+        const parentBlockEntity = getBlockAbove(editor, {
+          at: editor.selection!.anchor,
+        });
+
+        if (!getText(editor, parentBlockEntity![1])) {
+          // the selected block is empty
+          Transforms.removeNodes(editor, { at: editor.selection!.focus });
+
+          return true;
+        }
+
+        if (hasListChild(editor, listItem[0])) {
+          // the next block has children, so we have to move the first item up
+          const sublistRes = getListItemEntry(editor, {
+            at: [...listItem[1], 1, 0, 0],
```

Comment:
Is there a better way about this by any chance @zbeyens? Would be ideal if there was a common function, i think.

Reply:
Yeah, I hated to do this, maybe we should write a `getSublistEntry` or something that can be called after `hasListChild`